### PR TITLE
Allow to add, update picture from viewpoints

### PR DESCRIPTION
**Pull request by jrmi**
- Pictures can be edited from viewpoints
- Add `redirect` to routeur state to handle this case
- Picture can now be saved

### CODE REVIEW

File: src/modules/RA/Viewpoint/components/ViewpointFields.js [hunk 1]
```diff
@@ -73,7 +80,11 @@ const ViewpointFields = ({ edit, classes, mapConfig, ...props }) => {
 
         <TextInput source="properties.voie" />
 
-        {waiting && <><LinearProgress /></>}
+        {waiting && (
+          <>
+            <LinearProgress />
+          </>
```
Comment (SebCorbin) [inline src/modules/RA/Viewpoint/components/ViewpointFields.js]: Is this fragment really required?

File: src/modules/RA/Viewpoint/components/ViewpointFields.js [hunk 2]
```diff
@@ -97,67 +108,98 @@ const ViewpointFields = ({ edit, classes, mapConfig, ...props }) => {
...
+          <RichTextInput source="properties.dynamiques" />
+          <LongTextInput source="properties.issues" />
+          <LongTextInput source="properties.observations" />
+          <LongTextInput source="properties.historial-data" />
+          <LongTextInput source="properties.cultural-references" />
+
+          {waiting && (
+            <>
+              <LinearProgress />
+            </>
```
Comment (SebCorbin) [inline src/modules/RA/Viewpoint/components/ViewpointFields.js]: Same question: fragment really required?

File: src/modules/RA/Viewpoint/components/ViewpointFields.js (hunk 1 above)
Comment (jrmi) [reply src/modules/RA/Viewpoint/components/ViewpointFields.js]: Yes, to protect LinearProgress from ReactAdmin props automatically given to all childs

File: src/modules/RA/Viewpoint/components/GridListPictures.js [hunk 3]
```diff
@@ -48,35 +50,79 @@ const getColsForWidth = width => {
...
+                  {state}
                 </span>
-            )}
-              actionIcon={<IconButton><EditIcon color="secondary" /></IconButton>}
+              )}
+              actionIcon={(
+                <IconButton>
+                  <EditIcon color="secondary" />
+                </IconButton>
+)}
```
Comment (mabhub) [inline src/modules/RA/Viewpoint/components/GridListPictures.js]: Indentation to be corrected.

File: src/modules/RA/Viewpoint/components/ViewpointFields.js [hunk 4]
```diff
@@ -97,67 +108,98 @@ const ViewpointFields = ({ edit, classes, mapConfig, ...props }) => {
...
+            formClassName={classes.inline}
+          />
+          <SelectInput
+            source="properties.difficulty"
+            choices={[]}
+            formClassName={classes.inline}
+          />
+          <LongTextInput source="properties.note" rows={4} rowsMax={30} />
+        </FormTab>
       )}
```
Comment (mabhub) [inline src/modules/RA/Viewpoint/components/ViewpointFields.js]:
```jsx
{edit && (
  <FormTab>
  {/* ... */}
  </FormTab>
)}
{edit && (
  <FormTab>
  {/* ... */}
  </FormTab>
)}
{edit && (
  <FormTab>
  {/* ... */}
  </FormTab>
)}
```
with could be replaces by:
```jsx
{edit && (
  <FormTab>
  {/* ... */}
  </FormTab>
  <FormTab>
  {/* ... */}
  </FormTab>
  <FormTab>
  {/* ... */}
  </FormTab>
)}
```

**Comment by mabhub** [comment]
As an optionnal improvement, you could maybe try to split `Fix linting` commit content into smaller ones that you can fixup into each corresponding ones.

**Pull request merged**